httpie/downloads.py

```
@@ -0,0 +1,434 @@
+            # TODO: Should the filename be taken from response.history[0].url?
+            # Output file not specified. Pick a name that doesn't exist yet.
+            filename = None
+            if 'Content-Disposition' in response.headers:
+                filename = filename_from_content_disposition(
+                    response.headers['Content-Disposition'])
+            if not filename:
+                filename = filename_from_url(
+                    url=response.url,
+                    content_type=response.headers.get('Content-Type'),
+                )
+            self._output_file = open(get_unique_filename(filename), mode='a+b')
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)